src/main/scala/beam/utils/scenario/UrbanSimScenarioLoader.scala

```diff
@@ -565,13 +565,27 @@ class UrbanSimScenarioLoader(
           planInfo.legMode match {
             case Some(mode) =>
               val leg = PopulationUtils.createLeg(mode)
-              leg.getAttributes.putAttribute("trip_id", planInfo.tripId)
+
+              if (planInfo.tripId != null) {
```

Comment:
You can embed this:

```
leg.getAttributes.putAttribute("trip_id", if(planInfo.tripId != null) planInfo.tripId else "")
```

or (more terse, but more objects for GC)

```
leg.getAttributes.putAttribute("trip_id", Option(planInfo.tripId).getOrElse(""))
```

Reply:
Or since it is all off of `tripId`, then you could move it outside of the match to be used in a single place:

```
val tripId = //whichever path to set - if/else or Option
..
//use stored tripId directly
```